hpx/runtime/actions/continuation.hpp

```diff
@@ -925,6 +939,14 @@ namespace hpx { namespace actions
     template <typename Arg0>
     void continuation::trigger(Arg0 && arg0)
     {
+        // The dynamic cast decays the argument type to avoid the assert firing
+        // for cases when Arg0 is a const&. This does not make the code invalid
+        // as trigger_value (which is a virtual function) takes its argument
+        // by && anyways.
+        HPX_ASSERT(0 != dynamic_cast<
+                typed_continuation<typename util::decay<Arg0>::type> *
+            >(this));
+
```

Comment:
If you decay copy, everything works, here is the patch that works for me:
https://gist.github.com/sithhell/7c4f73280b236662e448b571a9218030